plugins/app-backend/src/lib/config/injectConfigIntoHtml.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+import { resolve as resolvePath } from 'path';
+import { InjectOptions } from './types';
+import compileTemplate from 'lodash/template';
+import { Config, ConfigReader } from '@backstage/config';
+
+const HTML_TEMPLATE_NAME = 'index.html.tmpl';
+
+/** @internal */
+export async function injectConfigIntoHtml(
+  options: InjectOptions,
+): Promise<boolean> {
+  const { rootDir, appConfigs } = options;
+
+  const templatePath = resolvePath(rootDir, HTML_TEMPLATE_NAME);
+
+  if (!(await fs.exists(templatePath))) {
+    return false;
+  }
+
+  const templateContent = await fs.readFile(
+    resolvePath(rootDir, HTML_TEMPLATE_NAME),
+    'utf8',
+  );
+
+  const config = ConfigReader.fromConfigs(appConfigs);
+
+  const templateSource = compileTemplate(templateContent, {
+    interpolate: /<%=([\s\S]+?)%>/g,
+  });
+
+  const publicPath = resolvePublicPath(config);
+  const indexHtmlContent = templateSource({
+    config,
+    publicPath,
+  });
+
+  const indexHtmlContentWithConfig = indexHtmlContent.replace(
```

Comment:
This one doesn't seem to have a mechanism for "if already present, replace instead of inject" like `injectConfigIntoStatic` does. Is it not needed since it always re-templates an original .tmpl file? Just double checking.

Reply:
Yep exactly, makes the injection a lot cleaner to not need to handle that